### PR TITLE
[feat]: [CDS-77114] : add override v2 adaptor in overrides v1 apis

### PR DESCRIPTION
**Pull request by tchaurasiya**
## Describe your changes

* Direct v1 override calls to v2 post converting v1 override request DTO to v2 requestDTO (calls : upsert+delete)
* Converting back v2 response DTO to v1 (calls - upsert+delete+list)
* Above is behaviour when FF ans Setting both are enabled

This is to support v1 type of api automation and terraform calls in v2 config.

 above calls will be applicable after all accounts will be migrated to v2. And only v2 type of overrides wil lremain

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/51690)

### CODE REVIEW

**Comment by tchaurasiya** [comment]
trigger uts

**Comment by tchaurasiya** [comment]
trigger ut0

File: 127-cd-nextgen-entities/src/main/java/io/harness/ng/core/serviceoverridev2/beans/ServiceOverrideRequestDTOV2.java [hunk 1]
```diff
@@ -52,4 +52,5 @@ public class ServiceOverrideRequestDTOV2 {
   @Schema(description = "Spec for overrides, containing overriding fields like manifests, variables, config files")
   ServiceOverridesSpec spec;
   String yamlInternal;
+  boolean v1Api;
```
Comment (lovish1999) [inline 127-cd-nextgen-entities/src/main/java/io/harness/ng/core/serviceoverridev2/beans/ServiceOverrideRequestDTOV2.java:55]: Is it required ?

**Review by lovish1999 — APPROVED** [review]

**Review by rafaelmart1ns — APPROVED** [review]

**Review by ivan-83 — APPROVED** [review]
:lgtm:

> Reviewed 12 of 12 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/51690#-:-NcxaxBg3ZzoIRSIrYP4:bnfp4nl)* status: 12 file(s) to review. 

___
*[`120-ng-manager/src/main/java/io/harness/ng/core/environment/resources/EnvironmentResourceV2.java` line 769 at r1](https://reviewable.io/reviews/harness/harness-core/51690#-NcxVm0Y99Est2-MIknB:-NcxVm0Y99Est2-MIknC:bqxio8u) ([raw file](https://github.com/harness/harness-core/blob/cc027d984c5f38eea8836358ef4cfa964937a35b/120-ng-manager/src/main/java/io/harness/ng/core/environment/resources/EnvironmentResourceV2.java#L769)):*
> ```Java
>         overridesEntity.getServiceRef());
>     ResponseDTO<ServiceOverridesResponseDTOV2> apiResponseV2 = null;
>     if (overrideEntityInDB.isPresent()) {
> ```

Could it be the case that some overrideEntityInDB property is present but missing in requestV2 for update?
___
*[`127-cd-nextgen-entities/src/main/java/io/harness/ng/core/serviceoverride/mapper/ServiceOverridesMapper.java` line 128 at r1](https://reviewable.io/reviews/harness/harness-core/51690#-NcxZ2N0BcGU4fXd2WyL:-NcxZ2N17DcuyljLz7Z0:b10ccs9) ([raw file](https://github.com/harness/harness-core/blob/cc027d984c5f38eea8836358ef4cfa964937a35b/127-cd-nextgen-entities/src/main/java/io/harness/ng/core/serviceoverride/mapper/ServiceOverridesMapper.java#L128)):*
> ```Java
>     String yamlFromV2Spec = EMPTY;
> 
>     if (serviceOverridesEntity.getSpec() != null && v2Enabled) {
> ```

could be the case that v2Enabled is not enabled, but serviceOverridesEntity.getSpec() exists?
___
*[`127-cd-nextgen-entities/src/main/java/io/harness/ng/core/serviceoverride/services/ServiceOverrideService.java` line 38 at r1](https://reviewable.io/reviews/harness/harness-core/51690#-NcxZRz2EOyU5R62ApHQ:-NcxZRz2EOyU5R62ApHR:bw0mqko) ([raw file](https://github.com/harness/harness-core/blob/cc027d984c5f38eea8836358ef4cfa964937a35b/127-cd-nextgen-entities/src/main/java/io/harness/ng/core/serviceoverride/services/ServiceOverrideService.java#L38)):*
> ```Java
>       String accountId, String orgIdentifier, String projectIdentifier, String environmentRef, String serviceRef);
> 
>   Optional<NGServiceOverridesEntity> getForV1AndV2(
> ```

Minor -  getForV1AndV2 >> getV1OrV2
___
*[`127-cd-nextgen-entities/src/main/java/io/harness/repositories/serviceoverride/spring/ServiceOverrideRepository.java` line 38 at r1](https://reviewable.io/reviews/harness/harness-core/51690#-Ncxa0rs6ZMBWlwKK_Mi:-Ncxa0rs6ZMBWlwKK_Mj:b-3dbmdm) ([raw file](https://github.com/harness/harness-core/blob/cc027d984c5f38eea8836358ef4cfa964937a35b/127-cd-nextgen-entities/src/main/java/io/harness/repositories/serviceoverride/spring/ServiceOverrideRepository.java#L38)):*
> ```Java
>   Optional<NGServiceOverridesEntity>
>   findByAccountIdAndOrgIdentifierAndProjectIdentifierAndEnvironmentRefAndServiceRefAndType(@NotEmpty String accountId,
>       String orgIdentifier, String projectIdentifier, @NotNull String environmentRef, String serviceRef,
> ```

Are enviromentRef and serviceRef mandatory? 
___
*[`127-cd-nextgen-entities/src/main/java/io/harness/repositories/serviceoverridesv2/spring/ServiceOverridesRepositoryV2.java` line 34 at r1](https://reviewable.io/reviews/harness/harness-core/51690#-Ncxame403k-wtLt-UKD:-Ncxame403k-wtLt-UKE:b-40dnv) ([raw file](https://github.com/harness/harness-core/blob/cc027d984c5f38eea8836358ef4cfa964937a35b/127-cd-nextgen-entities/src/main/java/io/harness/repositories/serviceoverridesv2/spring/ServiceOverridesRepositoryV2.java#L34)):*
> ```Java
>   Optional<NGServiceOverridesEntity>
>   getNGServiceOverridesEntityByAccountIdAndOrgIdentifierAndProjectIdentifierAndIdentifier(
>       String accountId, String orgIdentifier, String projectIdentifier, String identifier);
> ```

Minor - NotEmpty on accountId just for dev purposes

File: 127-cd-nextgen-entities/src/main/java/io/harness/ng/core/serviceoverridev2/beans/ServiceOverrideRequestDTOV2.java (hunk 1 above)
Comment (tchaurasiya) [reply 127-cd-nextgen-entities/src/main/java/io/harness/ng/core/serviceoverridev2/beans/ServiceOverrideRequestDTOV2.java:55]: no

**Comment by tchaurasiya** [comment]
To answer above comments, points needs to be consider is, above calls will be applicable after all accounts will be migrated to v2. And only v2 type of overrides will remain
**Could it be the case that some overrideEntityInDB property is present but missing in requestV2 for update?**
No. All properties are getting updated.

could be the case that v2Enabled is not enabled, but serviceOverridesEntity.getSpec() exists?
No, if (serviceOverridesEntity.getSpec() != null && v2Enabled) this snippet is just a safe guard that no v1 entity should be considered

Are enviromentRef and serviceRef mandatory?
enviromentRef is mandatory

File: 127-cd-nextgen-entities/src/main/java/io/harness/ng/core/serviceoverridev2/beans/ServiceOverrideRequestDTOV2.java (hunk 1 above)
Comment (yogesh-chauhan) [inline 127-cd-nextgen-entities/src/main/java/io/harness/ng/core/serviceoverridev2/beans/ServiceOverrideRequestDTOV2.java:55]: should this be hidden from open api ?

**Review by yogesh-chauhan — APPROVED** [review]

**Comment by tchaurasiya** [comment]
trigger review

**Pull request merged**